src/main/java/org/apache/sling/jcr/resource/internal/helper/AccessMetrics.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sling.jcr.resource.internal.helper;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *  PoJo to hold access metrics for a ResourceResolver; it is designed
+ *  for analysis of access patterns in lab situations; therefor its main
+ *  goal is to have very low overhead in case it is not used.
+ *  
+ *  this class is not threadsafe!
+ *  
+ *  This class implements autocloseable, so the statistics are written when the RR
+ *  is closed (see ResourceResolver.getPropertyMap)
+ *  
+ *
+ */
+public class AccessMetrics implements Closeable {
```

Comment:
I find the name a bit confusing. from the title i was expecting to see a usage of `org.apache.sling.commons.metrics.*` to  really write some metrics instead of log output.

so maybe calling it AccessLogger would better reflect what the class is doing.... unless you envision using metrics in the future, in which case it should be mentioned in the documentation.

Reply:
You are right, currently it's not more than a logger. Will rename it to ```AccessLogger```; if we change to expose real metrics, we can still rename it back again.